Sources/PointFree/EnvVars.swift

```diff
@@ -1,43 +1,116 @@
+  var baseUrlString = "http://localhost:8080"
+  var basicAuth = BasicAuth()
+  var gitHub = GitHub()
+  var mailgun = Mailgun()
+  var stripe = Stripe()
+
+  private enum CodingKeys: String, CodingKey {
+    case appSecret = "APP_SECRET"
+    case baseUrlString = "BASE_URL"
+  }
+
+  public struct Airtable: Codable {
+    var base1 = "deadbeef-base-1"
+    var base2 = "deadbeef-base-2"
+    var base3 = "deadbeef-base-3"
+    var bearer = "deadbeef-bearer"
+
+    private enum CodingKeys: String, CodingKey {
+      case base1 = "AIRTABLE_BASE_1"
+      case base2 = "AIRTABLE_BASE_2"
+      case base3 = "AIRTABLE_BASE_3"
+      case bearer = "AIRTABLE_BEARER"
+    }
+  }
+
+  public struct BasicAuth: Codable {
+    var username = "hello"
+    var password = "world"
+
+    private enum CodingKeys: String, CodingKey {
+      case username = "BASIC_AUTH_USERNAME"
+      case password = "BASIC_AUTH_PASSWORD"
+    }
+  }
+
+  public struct GitHub: Codable {
+    var clientId = "deadbeef-client-id"
+    var clientSecret = "deadbeef-client-secret"
+
+    private enum CodingKeys: String, CodingKey {
+      case clientId = "GITHUB_CLIENT_ID"
+      case clientSecret = "GITHUB_CLIENT_SECRET"
+    }
+  }
+
+  public struct Mailgun: Codable {
+    var apiKey = "deadbeef-mg-api-key"
+    var domain = "mg.domain.com"
+
+    private enum CodingKeys: String, CodingKey {
+      case apiKey = "MAILGUN_PRIVATE_API_KEY"
+      case domain = "MAILGUN_DOMAIN"
+    }
+  }
+
+  public struct Stripe: Codable {
+    var publishableKey = "pk_test"
+    var secretKey = "sk_test"
+
+    private enum CodingKeys: String, CodingKey {
+      case publishableKey = "STRIPE_PUBLISHABLE_KEY"
+      case secretKey = "STRIPE_SECRET_KEY"
+    }
+  }
+
+  public var baseUrl: URL {
+    return URL(string: self.baseUrlString)!
+  }
+
+  public static let `default` = { () -> EnvVars in
     let envFilePath = URL(fileURLWithPath: #file)
       .deletingLastPathComponent()
       .appendingPathComponent(".env")
 
     let localEnvVars = (try? Data(contentsOf: envFilePath))
       .flatMap { try? JSONDecoder().decode([String: String].self, from: $0) }
+      ?? [:]
 
-    return localEnvVars ?? ProcessInfo.processInfo.environment
-  }()
+    let envVars = localEnvVars.merging(ProcessInfo.processInfo.environment, uniquingKeysWith: { $1 })
 
-  static let appSecret = env["APP_SECRET"] ?? "deadbeefdeadbeefdeadbeefdeadbeef"
-  static let baseUrl = URL(string: env["BASE_URL"] ?? "http://localhost:8080")
-
-  enum Airtable {
-    static let base1 = env["AIRTABLE_BASE_1"] ?? "deadbeef-base-1"
-    static let base2 = env["AIRTABLE_BASE_2"] ?? "deadbeef-base-2"
-    static let base3 = env["AIRTABLE_BASE_3"] ?? "deadbeef-base-3"
-    static let bearer = env["AIRTABLE_BEARER"] ?? "deadbeef-bearer"
-  }
+    return (try? JSONSerialization.data(withJSONObject: envVars))
+      .flatMap { try? JSONDecoder().decode(EnvVars.self, from: $0) }
+      ?? EnvVars()
```

Comment:
This dance is a lil unfortunate.

Reply:
I guess another way to do this is to take the json config file that heroku gives us an then codegen EnvVars from it. then we'd have the compile time guarantees